### ISSUE TITLE
Expose an inventory of software versions of installed applications during the build process

### ISSUE THREAD

**Issue by cbgbt** [bottlerocket-os/bottlerocket#3269]
**What I'd like:**
After completing a Bottlerocket build, a file exists (perhaps under `build/images/$VARIANT/$tag/bottlerocket-$VARIANT-application-inventory.json`) which exposes version information of the installed software (e.g. "containerd is version 1.6.13")

Currently we [generate an application inventory in `rpm2img`](https://github.com/bottlerocket-os/bottlerocket/blob/develop/tools/rpm2img#L218) containing the manifest of software installed. This manifest contains information about the *Bottlerocket* version rather than the software versions, as its intended to be compared to the versioning information in [Bottlerocket security advisories](https://github.com/bottlerocket-os/bottlerocket/security/advisories). The published manifest is currently *only* included in the Bottlerocket image.

To sum up the two requirements:
* Bottlerocket's application inventory is modified to include versioning information about the installed software components themselves
* The application inventory is exposed by the build system as an artifact alongside the created OS image

**Any alternatives you've considered:**
The inventory published in the `build/` directory need not necessarily be the same as the one published within the host. This may be preferred if we would wish to avoid including additional fields in the currently published document.

**Comment by bcressey** [comment]
`application-inventory.json` is written alongside images as of https://github.com/bottlerocket-os/twoliter/pull/304/commits/61672b98d42bd2fc3b26d006762e3feb65cffb7e, and actual package versions are used inside the inventory since https://github.com/bottlerocket-os/twoliter/pull/384/commits/44a2623a6419dc7b07e346175a425bcad2f88aa6.
